syncer/heartbeat_test.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package syncer
+
+import (
+	"context"
+	"os"
+	"strconv"
+
+	. "github.com/pingcap/check"
+	"github.com/pingcap/dm/dm/config"
+)
+
+var _ = Suite(&testHeartbeatSuite{})
+
+type testHeartbeatSuite struct {
+	from config.DBConfig
+	lag  float64
+}
+
+func (t *testHeartbeatSuite) SetUpSuite(c *C) {
+	host := os.Getenv("MYSQL_HOST")
+	if host == "" {
+		host = "127.0.0.1"
+	}
+	port, _ := strconv.Atoi(os.Getenv("MYSQL_PORT"))
+	if port == 0 {
+		port = 3306
+	}
+	user := os.Getenv("MYSQL_USER")
+	if user == "" {
+		user = "root"
+	}
+	pswd := os.Getenv("MYSQL_PSWD")
+	t.from = config.DBConfig{
+		Host:     host,
+		User:     user,
+		Password: pswd,
+		Port:     port,
+	}
```

Comment:
can we extract these lines into a util function for test cases like `obtainMySQLTestConfig`?

Reply:
done